cpp/src/arrow/dataset/scanner.h

```diff
@@ -46,16 +49,13 @@ class ARROW_DS_EXPORT ScanOptions {
 
   // Filter
   std::shared_ptr<Expression> filter;
-
   // Evaluator for Filter
   std::shared_ptr<ExpressionEvaluator> evaluator;
 
-  // Schema to which record batches will be projected
+  // Schema to which record batches will be reconciled
   std::shared_ptr<Schema> schema;
-
-  std::vector<std::shared_ptr<FileScanOptions>> options;
-
-  bool include_partition_keys = true;
+  // Projecting the final RecordBatch to the requested schema.
```

Comment:
```suggestion
  // Projector for reconciling the final RecordBatch to the requested schema.
```

With this added, is `ScanOptions::schema` redundant? Will it ever be unequal to projector->schema()?